### PR TITLE
[OV JS] Remove acceptable_types parameter from js_to_cpp() template function

### PR DESCRIPTION
**Pull request by almilosz**
### Details:
 - Remove `const std::vector<napi_types>& acceptable_types` param from template `TargetType js_to_cpp(const Napi::Value&, const std::vector<napi_types>& acceptable_types);`
 - Remove template specialization `int32_t js_to_cpp<int32_t>(const Napi::CallbackInfo& info,
                           const size_t idx,
                           const std::vector<napi_types>& acceptable_types);`

### Tickets:
 - *140428*

### CODE REVIEW

File: src/bindings/js/node/src/helper.cpp [hunk 1]
```diff
@@ -136,45 +107,35 @@ std::unordered_set<std::string> js_to_cpp<std::unordered_set<std::string>>(
...
 
 template <>
-ov::element::Type_t js_to_cpp<ov::element::Type_t>(const Napi::CallbackInfo& info,
-                                                   const size_t idx,
-                                                   const std::vector<napi_types>& acceptable_types) {
+ov::element::Type_t js_to_cpp<ov::element::Type_t>(const Napi::CallbackInfo& info, const size_t idx) {
     const auto elem = info[idx];
-    if (!acceptableType(elem, acceptable_types))
-        OPENVINO_THROW(std::string("Cannot convert Napi::Value to ov::element::Type_t"));
+    OPENVINO_ASSERT(elem.IsString(), "Cannot convert Napi::Value to ov::element::Type_t");
```
Comment (vishniakov-nikolai) [inline src/bindings/js/node/src/helper.cpp:112]: What is about more specific message?
```suggestion
    OPENVINO_ASSERT(elem.IsString(), "Passed argument should be a String type");
```

File: src/bindings/js/node/src/helper.cpp [hunk 2]
```diff
@@ -106,14 +77,14 @@ std::vector<size_t> js_to_cpp<std::vector<size_t>>(const Napi::CallbackInfo& inf
         auto data_ptr = static_cast<int*>(buf.ArrayBuffer().Data());
         std::vector<size_t> vector(data_ptr, data_ptr + buf.ElementLength());
         return vector;
+    } else {
+        OPENVINO_THROW("Passed argument must be of type Array or TypedArray.");
     }
```
Comment (vishniakov-nikolai) [inline src/bindings/js/node/src/helper.cpp:82]: If else is last instruction in the function body we can white it out of condition
```suggestion
    }
    OPENVINO_THROW("Passed argument must be of type Array or TypedArray.");
```

File: src/bindings/js/node/src/helper.cpp [hunk 3]
```diff
@@ -184,7 +145,7 @@ ov::preprocess::ResizeAlgorithm js_to_cpp<ov::preprocess::ResizeAlgorithm>(
     } else if (algorithm == "RESIZE_LINEAR") {
         return ov::preprocess::ResizeAlgorithm::RESIZE_LINEAR;
     } else {
-        OPENVINO_THROW(std::string("Not supported resizeAlgorithm."));
+        OPENVINO_THROW("Not supported resizeAlgorithm.");
```
Comment (vishniakov-nikolai) [inline src/bindings/js/node/src/helper.cpp:148]: Possible to release from brackets

File: src/bindings/js/node/src/helper.cpp [hunk 4]
```diff
@@ -218,14 +179,11 @@ ov::Any js_to_cpp<ov::Any>(const Napi::Env& env, const Napi::Value& value) {
...
-    const Napi::CallbackInfo& info,
-    const size_t idx,
-    const std::vector<napi_types>& acceptable_types) {
+std::map<std::string, ov::Any> js_to_cpp<std::map<std::string, ov::Any>>(const Napi::CallbackInfo& info,
+                                                                         const size_t idx) {
     const auto elem = info[idx];
-    if (!acceptableType(elem, acceptable_types)) {
-        OPENVINO_THROW(std::string("Cannot convert Napi::Value to std::map<std::string, ov::Any>"));
-    }
+    OPENVINO_ASSERT(elem.IsObject(), "Cannot convert Napi::Value to std::map<std::string, ov::Any>");
```
Comment (vishniakov-nikolai) [inline src/bindings/js/node/src/helper.cpp:185]: Element at index **idx** should be an Object type

**Review by vishniakov-nikolai — APPROVED** [review]

File: src/bindings/js/node/src/helper.cpp (hunk 2 above)
Comment (almilosz) [reply src/bindings/js/node/src/helper.cpp:82]: I would like to keep it for the sake of readability

**Review by vishniakov-nikolai — APPROVED** [review]

**Pull request merged**